builtin/audit/socket/backend.go

```diff
@@ -112,19 +117,30 @@ func Factory(ctx context.Context, conf *audit.BackendConfig, useEventLogger bool
 	}
 	var w audit.Writer
 	switch format {
-	case "json":
+	case audit.JSONFormat.String():
 		w = &audit.JSONWriter{Prefix: conf.Config["prefix"]}
-	case "jsonx":
+	case audit.JSONxFormat.String():
 		w = &audit.JSONxWriter{Prefix: conf.Config["prefix"]}
 	}
 
+	formatterNodeID := event.GenerateNodeID()
+	b.nodeIDList = append(b.nodeIDList, formatterNodeID)
+	b.nodeMap[formatterNodeID] = f
 	fw, err := audit.NewEntryFormatterWriter(b.formatConfig, f, w)
 	if err != nil {
 		return nil, fmt.Errorf("error creating formatter writer: %w", err)
 	}
 
 	b.formatter = fw
 
+	sinkNode, err := event.NewSocketSink(format, address, event.WithSocketType(socketType), event.WithMaxDuration(writeDuration.String()))
+	if err != nil {
+		return nil, fmt.Errorf("error creating socket sink node: %w", err)
+	}
+	sinkNodeID := event.GenerateNodeID()
+	b.nodeIDList = append(b.nodeIDList, sinkNodeID)
+	b.nodeMap[sinkNodeID] = sinkNode
```

Comment:
Do we need these if we're not using the event broker?

Reply:
Good point! This decision to unconditionally create the eventlogger.Node instances was taken before we found that we needed to add the useEventLogger parameter to the Factory function. Now that we have that flag, we can use it to only create the nodes when we are using the event broker.